File: src/test/java/io/bloco/faker/components/TimeTest.java
package io.bloco.faker.components;

import io.bloco.faker.Faker;
import io.bloco.faker.helpers.Period;
import org.joda.time.DateTime;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.both;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.Matchers.greaterThanOrEqualTo;
import static org.hamcrest.Matchers.lessThanOrEqualTo;
import static org.junit.Assert.assertThat;

public class TimeTest {

  private Faker faker;

  @Before public void setUp() {
    faker = new Faker();
  }

  @Test public void between() {
    DateTime from = new DateTime("1990-06-22");
    DateTime to = new DateTime("1990-06-23");
    DateTime between = new DateTime(faker.time.between(from.toDate(), to.toDate()));

    assertThat(between.isAfter(from), is(true));
    assertThat(between.isBefore(to), is(true));
  }

  @Test public void forward() {
    DateTime now = DateTime.now();
    DateTime tenDaysLater = new DateTime(faker.time.forward(10));
    assertThat(tenDaysLater.getDayOfMonth(), is(now.plusDays(10).getDayOfMonth()));
  }

  @Test public void backward() {
    DateTime now = DateTime.now();
    DateTime tenDaysBefore = new DateTime(faker.time.backward(10));
    assertThat(tenDaysBefore.getDayOfMonth(), is(now.minusDays(10).getDayOfMonth()));
  }

  @Test public void allPeriod() {
    DateTime allPeriod = new DateTime(faker.time.forward(1, Period.all));
    assertThat(allPeriod.getHourOfDay(),
        is(both(greaterThanOrEqualTo(0)).and(lessThanOrEqualTo(23))));
  }

  @Test public void dayPeriod() {
    DateTime dayPeriod = new DateTime(faker.time.forward(1, Period.day));
    assertThat(dayPeriod.getHourOfDay(),
        is(both(greaterThanOrEqualTo(9)).and(lessThanOrEqualTo(17))));
  }

  @Test public void nightPeriod() {
    DateTime nightPeriod = new DateTime(faker.time.forward(1, Period.night));
    assertThat(nightPeriod.getHourOfDay(),
        is(both(greaterThanOrEqualTo(18)).and(lessThanOrEqualTo(23))));
  }

  @Test public void morningPeriod() {
    DateTime morningPeriod = new DateTime(faker.time.forward(1, Period.morning));
    assertThat(morningPeriod.getHourOfDay(),
        is(both(greaterThanOrEqualTo(6)).and(lessThanOrEqualTo(11))));
  }

  @Test public void afternoonPeriod() {
    DateTime afternoonPeriod = new DateTime(faker.time.forward(1, Period.afternoon));
    assertThat(afternoonPeriod.getHourOfDay(),
        is(both(greaterThanOrEqualTo(12)).and(lessThanOrEqualTo(17))));
  }

  @Test public void eveningPeriod() {
    DateTime eveningPeriod = new DateTime(faker.time.forward(1, Period.evening));
    assertThat(eveningPeriod.getHourOfDay(),
        is(both(greaterThanOrEqualTo(17)).and(lessThanOrEqualTo(21))));
  }

  @Test public void midnightPeriod() {
    DateTime midnightPeriod = new DateTime(faker.time.forward(1, Period.midnight));
    assertThat(midnightPeriod.getHourOfDay(),
        is(both(greaterThanOrEqualTo(0)).and(lessThanOrEqualTo(4))));
  }
}

File: src/test/kotlin/io/bloco/faker/components/TimeTest.kt
package io.bloco.faker.components

import io.bloco.faker.Faker
import io.bloco.faker.helpers.Period
import io.bloco.faker.helpers.toLocalDateTime
import junit.framework.TestCase.assertTrue
import org.junit.Test
import java.text.SimpleDateFormat
import java.time.LocalDateTime

class TimeTest {
    private val faker = Faker()
    private val formatter = SimpleDateFormat("yyyy-MM-dd")

    @Test
    fun between() {
        val from = formatter.parse("1990-06-22")
        val to = formatter.parse("1990-06-23")
        val between = faker.time.between(from, to).toLocalDateTime()

        assertTrue(between.isAfter(from.toLocalDateTime()))
        assertTrue(between.isBefore(to.toLocalDateTime()))
    }

    @Test
    fun forward() {
        val now = LocalDateTime.now()
        val tenDaysLater = faker.time.forward(10).toLocalDateTime()
        assertTrue(tenDaysLater.dayOfMonth == now.plusDays(10).dayOfMonth)
    }

    @Test
    fun backward() {
        val now = LocalDateTime.now()
        val tenDaysBefore = faker.time.backward(10).toLocalDateTime()
        assertTrue(tenDaysBefore.dayOfMonth == now.minusDays(10).dayOfMonth)
    }

    @Test
    fun allPeriod() {
        val allPeriod = faker.time.forward(1, Period.All).toLocalDateTime()
        assertTrue(allPeriod.hour in 0..23)
    }

    @Test
    fun dayPeriod() {
        val dayPeriod = faker.time.forward(1, Period.Day).toLocalDateTime()
        assertTrue(dayPeriod.hour in 9..17)
    }

    @Test
    fun nightPeriod() {
        val nightPeriod = faker.time.forward(1, Period.Night).toLocalDateTime()
        assertTrue(nightPeriod.hour in 18..23)
    }

    @Test
    fun morningPeriod() {
        val morningPeriod = faker.time.forward(1, Period.Morning).toLocalDateTime()
        assertTrue(morningPeriod.hour in 6..11)
    }

    @Test
    fun afternoonPeriod() {
        val afternoonPeriod = faker.time.forward(1, Period.Afternoon).toLocalDateTime()
        assertTrue(afternoonPeriod.hour in 12..17)
    }

    @Test
    fun eveningPeriod() {
        val eveningPeriod = faker.time.forward(1, Period.Evening).toLocalDateTime()
        assertTrue(eveningPeriod.hour in 17..21)
    }

    @Test
    fun midnightPeriod() {
        val midnightPeriod = faker.time.forward(1, Period.Midnight).toLocalDateTime()
        assertTrue(midnightPeriod.hour in 0..4)
    }
}
